### PR TITLE
all: where appropriate update examples to compile but not run, and add tests where needed for coverage

### LINKED ISSUE #869: pubsub: each cloud driver implementation should have examples
The examples don't have to demonstrate how to use the concrete type; the concrete type examples should show that. The per-provider examples are for demonstrating how to use the provider-specific constructor.

For example, `gcppubsub` has:

```
func OpenTopic(ctx context.Context, client *raw.PublisherClient, proj gcp.ProjectID, topicName string, opts *TopicOptions) (*pubsub.Topic, error) {
```

An example would make it easier for a new user to figure out how to create a `raw.PublisherClient`.

### LINKED ISSUE #1177: all: update non-local examples to compile but not run instead of expecting failure
The examples should ideally be fairly realistic and easy to copy and paste into real apps without too much modification. The output should show things working, not failures due to not being able to assume the example can access the network.

That's not the current situation though. For example, in `blob/gcsblob/example_test.go` the [expected output](https://github.com/google/go-cloud/blob/d22bb640872cc897bb38b124f1ce8c55197fda27/blob/gcsblob/example_test.go#L70) is 

> ReadAll failed due to invalid credentials.

One possible solution would be to put a line like this at the top of the example_test.go files:
```
// +build !no_network
```
and then pass `-tag=no_network` to `go test` within Travis. A downside would be that these examples would not be built or run from Travis, though presumably we would find out quickly if they are broken since people would try them and report any issues they find. 

This solution isn't perfect, but I think it beats the current situation where our examples don't really make sense.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #1177.
Fixes #869.

### CODE REVIEW

File: blob/azureblob/example_test.go [hunk 1]
```diff
@@ -42,232 +39,71 @@ func Example() {
...
-				}
-			}
-		}
-	}
-
-	// Output:
-	// ReadAll failed due to invalid credentials
+	b, err = blob.Open(ctx, "azblob://mycontainer?cred_path=replace-with-path-to-credentials-file")
+	_ = b
+	_ = err
```
Comment (shantuo) [inline blob/azureblob/example_test.go]: nit: err can be moved before `_ = b` or ignored entirely.

File: pubsub/mempubsub/example_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2018 The Go Cloud Development Kit Authors
```
Comment (shantuo) [inline pubsub/mempubsub/example_test.go]: s/2018/2019

File: pubsub/rabbitpubsub/example_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 The Go Cloud Development Kit Authors
```
Comment (shantuo) [inline pubsub/rabbitpubsub/example_test.go]: s/2018/2019

File: blob/azureblob/example_test.go (hunk 1 above)
Comment (vangent) [reply blob/azureblob/example_test.go]: Sorry, I didn't get this. I need to touch both b and err or else there's a compile error that they aren't used.

File: pubsub/mempubsub/example_test.go (hunk 2 above)
Comment (vangent) [reply pubsub/mempubsub/example_test.go]: Done.

File: pubsub/rabbitpubsub/example_test.go (hunk 3 above)
Comment (vangent) [reply pubsub/rabbitpubsub/example_test.go]: Done.

**Pull request merged**